src/Documentation/Event-Sourcing/LogConsistencyProviders.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: Included Log-Consistency Providers
+---
+
+# Built-In Log-Consistency Providers
+
+The `Microsoft.Orleans.EventSourcing` package includes several log-consistency providers that cover basic scenarios suitable to get started, and allow some extensibility.
+
+
+### Orleans.EventSourcing.**StateStorage**.LogConsistencyProvider
+
+This provider stores *grain state snapshots*, using a standard storage provider that can be configured independently. 
+
+The data that is kept in storage is an object that contains both the grain state (as specified by the first type parameter to `JournaledGrain`) and some meta-data (the version number, amd a special tag that is used to avoid duplication of events when storage accesses fail).
+
+Since the entire grain state is read/written every time we access storage, this provider is not suitable for objects whose grain state is very large. 
+
+This provider does not support `RetrieveConfirmedEvents`: it cannot retrieve the events from storage because the events are not persisted.
```

Comment:
> About the snapshotting: we probably need a separate discussion on that, and write some higher-quality provider that actually does it. I never quite understood the details of how this is supposed to work. 

IMHO snapshot is not something that the provider must care about. The current snapshot is the current Grain State object with all the events already applied. If someone need to push this snapshot to somewhere else, just send the object to it. There is nothing the provider should do about it.

> After it fails to successfully deserialize the snapshot

Like I said, we shouldn't serialize the snapshot. It is a transient state in memory of all the events applied to it.

> How does the provider know when it has to replay the log because the transitions have changed?

The events should be replay before `OnActivateAsync()` and applied to the in-memory State object so it can build the current Snapshot of the state. 

Whenever an even is raised, it is applied by the provider into the State object, stored into the event store (not regular grain state storage) and optionally a callback implemented into the grain code can be fired (i.e. `OnEventRaised()`) so the grain can for example publish this event to a stream so other grains can consume and react to it (i.e. to generate projections).